sky/packages/sky/lib/src/material/dropdown.dart

```diff
@@ -0,0 +1,258 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' as ui;
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'icon.dart';
+import 'ink_well.dart';
+import 'shadows.dart';
+import 'theme.dart';
+
+const Duration _kMenuDuration = const Duration(milliseconds: 300);
+const double _kMenuItemHeight = 48.0;
+const double _kMenuHorizontalPadding = 36.0;
+const double _kBaselineOffsetFromBottom = 20.0;
+const Border _kDropdownUnderline = const Border(bottom: const BorderSide(color: const Color(0xFFBDBDBD), width: 2.0));
+
+class _DropdownMenu extends StatelessComponent {
+  _DropdownMenu({
+    Key key,
+    this.items,
+    this.rect,
+    this.performance,
+    this.selectedIndex,
+    this.level: 4
+  }) : super(key: key) {
+    assert(items != null);
+    assert(performance != null);
+  }
+
+  final List<DropdownMenuItem> items;
+  final Rect rect;
+  final PerformanceView performance;
+  final int selectedIndex;
+  final int level;
+
+  Widget build(BuildContext context) {
+    // The menu is shown in three stages (unit timing in brackets):
+    // [0 - 0.25] - Fade in a rect-sized menu container with the selected item.
+    // [0.25 - 0.5] - Grow the otherwise empty menu container from the center until it's
+    //   big enough for as many items as we're going to show.
+    // [0.5 - 1.0] Fade in the remaining visible items from top to bottom.
+    //
+    // When the menu is dismissed we do our best to just fade the entire thing out
+    // in the first 0.25.
+
+    final double unit = 0.5 / (items.length + 1.5);
+    final List<Widget> children = <Widget>[];
+    for (int itemIndex = 0; itemIndex < items.length; ++itemIndex) {
+      AnimatedValue<double> opacity;
+      if (itemIndex == selectedIndex) {
+        opacity = new AnimatedValue<double>(0.0, end: 1.0, curve: const Interval(0.0, 0.001), reverseCurve: const Interval(0.75, 1.0));
+      } else {
+        final double start = (0.5 + (itemIndex + 1) * unit).clamp(0.0, 1.0);
+        final double end = (start + 1.5 * unit).clamp(0.0, 1.0);
+        opacity = new AnimatedValue<double>(0.0, end: 1.0, curve: new Interval(start, end), reverseCurve: const Interval(0.75, 1.0));
+      }
+      children.add(new FadeTransition(
+        performance: performance,
+        opacity: opacity,
+        child: new InkWell(
+          child: items[itemIndex],
+          onTap: () {
+          Navigator.of(context).pop(items[itemIndex].value);
```

Comment:
I'll indent. I think the Dart style guide advises against using the fat arrow when there's no return value.